pkg/provisioner/ironic/delete_test.go

```diff
@@ -190,3 +219,100 @@ func TestDelete(t *testing.T) {
 		})
 	}
 }
+
+func TestDeleteWithPowerOff(t *testing.T) {
+
+	nodeUUID := "33ce8659-7400-4c68-9535-d10766f07a58"
+
+	cases := []struct {
+		name        string
+		ironic      *testserver.IronicMock
+		inspector   *testserver.InspectorMock
+		priorErrors int
+
+		expectedPowerState string
+		expectedError      string
+	}{
+		{
+			name: "test-power-change-without-errors",
+			ironic: testserver.NewIronic(t).Node(
+				nodes.Node{
+					UUID:           nodeUUID,
+					ProvisionState: "active",
+					Maintenance:    true,
+					PowerState:     powerOn,
+				},
+			).WithNodeStatesPowerUpdate(nodeUUID, http.StatusAccepted),
+			priorErrors:        0,
+			expectedPowerState: "power off",
```

Comment:
This is more a `expectedPowerStateRequestChangeTo` (or something less verbose, the point here is that this is a request more than an expected result)